diff --git a/org/eclipse/che/ide/editor/orion/public/orion-8.0/built-editor-amd.js b/org/eclipse/che/ide/editor/orion/public/orion-8.0/built-editor-amd.js
index d81f7e1..81b2cbc 100644
--- a/org/eclipse/che/ide/editor/orion/public/orion-8.0/built-editor-amd.js
+++ b/org/eclipse/che/ide/editor/orion/public/orion-8.0/built-editor-amd.js
@@ -1,10 +1,10 @@
 /* orion editor */ /*******************************************************************************
  * @license
  * Copyright (c) 2014 IBM Corporation and others.
- * All rights reserved. This program and the accompanying materials are made 
- * available under the terms of the Eclipse Public License v1.0 
- * (http://www.eclipse.org/legal/epl-v10.html), and the Eclipse Distribution 
- * License v1.0 (http://www.eclipse.org/org/documents/edl-v10.html). 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0
+ * (http://www.eclipse.org/legal/epl-v10.html), and the Eclipse Distribution
+ * License v1.0 (http://www.eclipse.org/org/documents/edl-v10.html).
  *
  * Contributors:
  *     IBM Corporation - initial API and implementation
@@ -7055,6 +7055,7 @@ define("orion/editor/textView", [  //$NON-NLS-0$
 		_handleMouseWheel: function (e) {
 			if (this._noScroll) return;
 			var lineHeight = this._getLineHeight();
+			if(e.defaultPrevented) return;
 			var pixelX = 0, pixelY = 0;
 			// Note: On the Mac the correct behaviour is to scroll by pixel.
 			if (util.isIE || util.isOpera) {
@@ -37485,6 +37486,7 @@ define('orion/editor/edit', [ //$NON-NLS-0$
 			annotationFactory: new mEditorFeatures.AnnotationFactory(),
 			lineNumberRulerFactory: new mEditorFeatures.LineNumberRulerFactory(),
 			foldingRulerFactory: new mEditorFeatures.FoldingRulerFactory(),
+			zoomRulerFactory: new mEditorFeatures.ZoomRulerFactory(),
 			textDNDFactory: new mEditorFeatures.TextDNDFactory(),
 			contentAssistFactory: contentAssistFactory,
 			keyBindingFactory: new mEditorFeatures.KeyBindingsFactory(), 
