diff --git a/addon/edit/closetag.js b/addon/edit/closetag.js
index 369bea3..e68d52d 100644
--- a/addon/edit/closetag.js
+++ b/addon/edit/closetag.js
@@ -131,7 +131,7 @@
 
   function autoCloseSlash(cm) {
     if (cm.getOption("disableInput")) return CodeMirror.Pass;
-    autoCloseCurrent(cm, true);
+    return autoCloseCurrent(cm, true);
   }
 
   CodeMirror.commands.closeTag = function(cm) { return autoCloseCurrent(cm); };